### PR TITLE
[ubuntu-dev] force pip to use python 2.7 and not python 3+

### PR DESCRIPTION
**Pull request by jankeromnes**
This is terrible. Please suggest a better solution if you know one.

### CODE REVIEW

**Comment by jankeromnes** [comment]
By the way this is to fix [Kresus builds](https://hub.docker.com/r/bnjbvr/kresus-janitor/builds/), but the regression affects any other image that uses `pip install` while expecting Python 2.7.

**Comment by jankeromnes** [comment]
@bnjbvr Please have a look if you can. 😄

**Comment by bnjbvr** [comment]
This is quite unfortunate. It would be more verbose, but would make it sense to apply the strategy advised in https://stackoverflow.com/questions/2812520/pip-dealing-with-multiple-python-versions ? i.e. use `pip2` or `pip3` whenever it's needed in place of `pip`.

**Comment by jankeromnes** [comment]
> This is quite unfortunate.

I agree. 😕

> i.e. use `pip2` or `pip3` whenever it's needed in place of `pip`.

This does seem like the ideal solution, however this only works if you control every call-site of `pip`. That's not the case for Janitor: Even if we could replace all references to `pip` with `pip2` (e.g. in Kresus' install scripts), there will probably be other dependencies or software projects that will still use `pip` and expect it to be Python 2.7 (e.g. legacy software projects).

I read somewhere online that `pip` should use the same Python version as `python`. However, in current Janitor environments it seems that `pip` uses 3+ while `python` is 2.7. Thus I would like to fix `pip` to correctly reference 2.7 (as `python` does, because that's what legacy software will expect), and count on more modern software to explicitely use `pip2` / `pip3` and `python2` / `python3`.

File: ubuntu-dev/ubuntu-dev.dockerfile [hunk 1]
```diff
@@ -61,6 +61,7 @@ RUN __LLVM_VERSION__="5.0" \
  && pip install --upgrade pip \
  && pip install --upgrade virtualenv \
  && pip3 install --upgrade pip \
```
Comment (Martiusweb) [inline ubuntu-dev/ubuntu-dev.dockerfile]: what if you do that before pip install --upgrade pip?

Comment (jankeromnes) [reply ubuntu-dev/ubuntu-dev.dockerfile]: Actually I don't think this solves the problem (`pip install --upgrade pip` says `Requirement already up-to-date: pip in /usr/local/lib/python2.7/dist-packages` and leaves `/usr/local/bin/pip` on Python 3.5).

**Comment by jankeromnes** [comment]
@bnjbvr @Martiusweb Thanks a lot for your input! However, it seems that we couldn't find cleaner solutions than the one I'm proposing, so I think we'll have to go ahead with the binary-overwrite fix for now. Any objections?

**Comment by jankeromnes** [comment]
ETIMEDOUT

Let's ship this as is.

**Pull request merged**